src/core/classes/factory.ts

```diff
@@ -120,6 +122,35 @@ export class ContractFactory extends ContractWrapper<TWFactory> {
           erc721metadata.platform_fee_basis_points,
           erc721metadata.platform_fee_recipient,
         ];
+      case SignatureDrop.contractType:
+        // Deploy a SigMint Contract
+        const signer = this.getSigner();
+        invariant(signer);
+        const sigMintDeployer = await new ethers.ContractFactory(
+          SigMint__factory.abi,
+          SigMint__factory.bytecode,
+        )
+          .connect(signer)
+          .deploy();
+
+        await sigMintDeployer.deployTransaction.wait();
```

Comment:
@nkrishang this requires users to do 2 transactions when deploying a SignatureDrop contract from the dashboard/sdk. I think we can avoid this by just pre-deploying the `SigMint` contract on every chain. It should be ok to use the `SigMint` same contract for any number of `SignatureDrop` contracts right?